internal/actions/update.go

```diff
@@ -181,58 +191,58 @@ func restartContainersInSortedOrder(containers []types.Container, client contain
 			continue
 		}
 		if stoppedImages[c.SafeImageID()] {
-			if err := restartStaleContainer(c, client, params); err != nil {
+			if err := us.restartContainer(c); err != nil {
 				failed[c.ID()] = err
-			} else if c.IsStale() {
+			} else if c.IsMarkedForUpdate() {
 				// Only add (previously) stale containers' images to cleanup
 				cleanupImageIDs[c.ImageID()] = true
```

Comment:
To me, rather than at line 89, this is a better place to call

			us.progress.MarkForUpdate(c.ID())

That would avoid setting a container's state to UpdatedState before it's actually updated. That would also avoid an item going from UpdatedState to FailedSate in cases where something fails (or if something later breaks with properly setting FailedState, misrepresenting the item as having been updated if it wasn't). This is probably not a big deal but I mention it because it was confusing to me initially why we're setting something to (past-tense) UpdatedState before any action is completed.